### PR TITLE
VEGA-2066 Create a draft LPA from a donor record #minor

### PR DESCRIPTION
**Pull request by MishNajam**
## Checklist

- [x] I have updated the end-to-end tests to work with my changes
- [ ] I have added styling for Dark Mode
- [ ] I have checked that my UI changes meet accessibility standards

### CODE REVIEW

File: internal/sirius/create_additional_draft_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+						Body: []map[string]interface{}{
+							{
+								"caseSubtype": dsl.String("personal-welfare"),
+								"uId":         dsl.Regex("M-GHIJ-7890-KLMN", `^M(-[0-9A-Z]{4}){3}$`),
+							},
+						},
+					})
+			},
+			expectedResponse: map[string]string{
+				"personal-welfare": "M-GHIJ-7890-KLMN",
```
Comment (gregtyler) [inline internal/sirius/create_additional_draft_test.go:57]:
```suggestion
				"caseSubtype": "personal-welfare",
				"uId": "M-GHIJ-7890-KLMN",
```

File: internal/sirius/create_additional_draft_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	}{
+		{
+			name: "Minimal",
+			additionalDraftData: AdditionalDraft{
+				CaseType: []string{"personal-welfare"},
+			},
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("I am a Modernised LPA user").
```
Comment (gregtyler) [inline internal/sirius/create_additional_draft_test.go:33]: I think this needs to be:
```suggestion
					Given("A donor exists").
```
To ensure that someone with ID 188 is already present

File: web/template/layout/mlpa-header.gohtml [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+        </span>
+        <div class="govuk-grid-column-one-quarter">
+            <div class="moj-button-menu" data-module="right-aligned-app-button-menu">
+                <div class="moj-button-menu__wrapper">
+                    <a href="{{ prefix (printf "/create-additional-draft-lpa?id=%d" .CaseSummary.DigitalLpa.SiriusData.Donor.ID) }}" role="button" draggable="false" class="govuk-button moj-button-menu__item govuk-button--secondary " data-module="dropdown-menu">
+                        Create a draft LPA
+                    </a>
+                </div>
+            </div>
+        </div>
```
[the remark this answers is not in the document]
Comment (MishNajam) [reply web/template/layout/mlpa-header.gohtml]: Tried using this and weirdly it ended up being slightly more to the left vs using the columns. Will leave in as is for now and play with CSS some more in a future ticket 🤔

File: web/assets/donor-record-dropdown-menu.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+      rightDropDownMenuToggle.addEventListener("change", (e) => {
+        const url = encodeURI(e.target.value);
+        if (url.startsWith("//") || !url.startsWith("/")) {
+          return;
+        }
+
+        window.location.href = url;
+      });
+    });
```
[the remark this answers is not in the document]
Comment (MishNajam) [reply web/assets/donor-record-dropdown-menu.js]: Need this for the donor actions dropdown menu included as part of the design:
- Oddly, the HTML suggested [here](https://design-patterns.service.justice.gov.uk/components/button-menu/) did not give the dropdown effect
- Also, reusing the existing dropdown-menu.js component lead to funky software gore where interacting with one would also use the other. 

I've renamed the component to something that makes more sense for now.

[the remark this answers is not in the document]
Comment (MishNajam) [reply web/assets/donor-record-dropdown-menu.js]: Ah, it's not used. Removed this now.

**Review by gregtyler — APPROVED** [review]

**Pull request merged**